### ISSUE TITLE
Support for custom ONNX optimized models

### ISSUE THREAD

**Issue by sujithjoseph** [MaartenGr/KeyBERT#48]
How can i pass a custom model url (say one optimized for inference for onnx / tensorRT).

**Comment by MaartenGr** [comment]
This is currently not supported as `Flair`, which is the main method for using embedding models, uses hugginface transformers which is typically not meant for importing onnx structures in part due to their tokenization structure. Exporting to onnx is possible however.

**Comment by sujithjoseph** [comment]
This is available in sentencetransformers - https://github.com/UKPLab/sentence-transformers/blob/72da7293186700bff4fed8e81adcfe40a03be1d2/examples/onnx_inference/onnx_inference.ipynb . We can load an onnx model for inference. With a local path to onnx exported file, it could be added as one more backend . Did test it with tinyBERT model, but didnt see much performance gain on a V100. yet to try on a T4 GPU.

**Comment by MaartenGr** [comment]
Thank you for the link! I did not realize it was already available in sentence-transformers. The main difficulty though will be loading in custom ONNX models that require some processing before it can be used in sentence-transformers.

**Comment by sujithjoseph** [comment]
I am thinking of adding it as a custom backend. i didnt see this before - https://maartengr.github.io/KeyBERT/guides/embeddings.html#custom-backend

**Comment by MaartenGr** [comment]
Yes, if you can create a custom backend that loads in using the sentence-transformer then that would solve your issue. It would also be nice if that backend could then be added to KeyBERT but that would also depend on dependencies and implementation.

**Comment by Matthieu-Tinycoaching** [comment]
Hi @MaartenGr @sujithjoseph did you manage to deal with ONNX backend?
